plaidml2/core/ffi.cc

```diff
@@ -67,26 +67,27 @@ void plaidml_init(plaidml_error* err) {
       auto eventlog_str = vertexai::env::Get("PLAIDML_EVENTLOG_FILENAME");
       if (eventlog_str.size()) {
         IVLOG(1, "Logging events to " << eventlog_str);
-        EventLog e;
-        e.set_filename(eventlog_str);
-        auto eventlog = EventLogFactory().MakeTypedInstance(*ctx, e);
+        vertexai::eventing::file::proto::EventLog e_config;
+        e_config.set_filename(eventlog_str);
+        auto eventlog = std::make_shared<vertexai::eventing::file::EventLog>(e_config);
         ctx->set_eventlog(std::move(eventlog));
         ctx->set_is_logging_events(true);
-      } else {
-        ctx->set_is_logging_events(false);
-        ctx->set_eventlog(std::shared_ptr<vertexai::context::EventLog>());
+        GetPlatform();
       }
-      GetPlatform();
     });
   });
 }
 
+void plaidml_reset_eventlog(void) {
+  auto ctx = GlobalContext::getContext();
+  ctx->set_eventlog(nullptr);
+}
+
 void plaidml_shutdown(plaidml_error* err) {
   ffi_wrap_void(err, [&] {
     IVLOG(1, "plaidml_shutdown");
     GetPlatform().platform.reset(nullptr);
-    auto ctx = GlobalContext::getContext();
-    ctx->eventlog()->FlushAndClose();
+    std::atexit(plaidml_reset_eventlog);
```

Comment:
Actually, we'll want to register the atexit call in `init`, so that it gets called on exit. We wouldn't want to register this on shutdown...